pkg/client/didexchange/client.go

```diff
@@ -17,6 +17,7 @@ import (
 
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/dispatcher"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/didexchange"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/didexchange/models"
```

Comment:
why was a new package needed?

Reply:
To avoid cyclic dependency... 
didexchange --> persistence --> didexchange

Reply:
Maybe remind me again why persistence had to be moved out?

You could use an interface and inject the ConnectionRecorder when instantiating the protocol service (for example).

IMO, the code looks a bit less readable with models.Invitation rather than didexchange.Invitation.

Reply:
Moved Invitation struct back to original package and moved connection records from `persistence` package to `didexchange` service package.